### ISSUE TITLE
Final Peer Review - jzc8

### ISSUE THREAD

**Issue by jimmychen623** [Archerhaha/LOLGameWinPrediction#13]
The objective of this project is to predict whether a League of Legends team will win a match given statistics about the match such as kill statistics or champion choice. The project also wanted to investigate the set of indicative features for winning a match.

Overall, the report was clean, well structured and sufficiently comprehensive. The team makes good use of visuals such as bar charts and frequency maps during their data visualization section and these were quite helpful in understanding how the data was distributed. The team makes use of several methods. Proximal gradient descent with various types of regularization and different loss functions was used. These were described with sufficient detail. The team also mentioned using classification decision trees, random forest, and SVMs with less detail.

I would like to give some constructive feedback on how the team could have done some things differently. 

Although this is not your fault, the dataset was not very comprehensive and completely clean, leaving little room for feature engineering and no room for data cleaning. In addition, the features related to the champions chosen and banned by each team are ignored and not used at all. There is no justification provided for this decision so I assume it was because of the categorical nature of the champions that led to this decision. I would have suggested one-hot encoding the champions and then autoencoding the one-hot vectors in order to pick out correlations and reduce the huge dimensionality spike that one-hot encoding will result in with this many classes. 

It would have been helpful to explore the feature space more. A lot of these variables seem correlated. For example, we'd like to know whether the team that gets first blood is more likely to get the first baron kill as the result of a "snowball" effect. Exploring feature correlations would have been useful in your analysis. This correlation also brings up the fact that the feature set is not representative of the state of before the game begins, but rather some features such as firstBaron occur much later. So there may be some dependency between features since the events they represent occur over the course of the game.

I also noticed that there was no mention of any data normalization. Data normalization is important when performing regression because the scale of numerical features can have a large impact on the weight coefficients chosen by the model. I noticed that the tower kills feature had the highest weight. This is likely because there are more tower kills than any other type of kill so the model favors it as a good predictive feature even though I would intuitively suspect that inhibitor kills are more indicative due to the resulting Super Minions and the fact that the enemy even got so close to the base (I don't play LoL, I just did some research so this is just my intuition). I suspect that if you normalized the features before running regression, you'd find that the model doesn't pick tower kills as the predominant indicator.

There was good usage of models but a general lack of explanation for why those models were chosen. For example, what led you to believe that a classification tree would be a good model choice or why you felt that the data had a linearly separable nature that made it fit for using a SVM.

Lastly, I noticed that the data was not correctly split up. You perform 5-fold cross-validation, but you train on 4 of the folds and use the remaining fold as **test**. What should have been done was to initially split the entire dataset randomly into 80% train and 20% test. The cross validation should be performed only on the training set to find optimal model parameters. The test set should not be opened. Then once you're satisfied, evaluate the model on your test set once and report those results. The way it is now, you end up training on the test data, which may invalidate the results.
